### ISSUE TITLE
Add --fix option

### ISSUE THREAD

**Issue by cjoudrey** [cjoudrey/graphql-schema-linter#23]
Some rules could be fixed by the linter for example `enum-values-sorted-alphabetically`.

It might be interesting to include a `--fix` option which would automatically resolve errors that can be fixed.

**Comment by bbohen** [comment]
@cjoudrey I'd like to check this out, I might not be able to get to it for a few days as I'm at the summit but I'll start messing around with it. Cool idea by the way 👍.

**Comment by cjoudrey** [comment]
That would be awesome @bbohen. 😀

Whenever you do poke at it, let me know if you need help or want to discuss solutions.

**Comment by bbohen** [comment]
Absolutely! I think it would be really nice to be use this with ESLint per #21.

**Comment by iamclaytonray** [comment]
@cjoudrey @bbohen - Is this still being worked on? If not, I may take a stab at it

**Comment by cjoudrey** [comment]
I’m not working on it atm. If @bbohen isn’t, I’d say go for it. 😀

**Comment by iamclaytonray** [comment]
@cjoudrey - I'll wait to see what @bbohen says

**Comment by bbohen** [comment]
@iamclaytonray Go for it! I was all excited to work on this and then work got really busy 😬 .

**Comment by iamclaytonray** [comment]
Thanks @bbohen!

No promises but I'll try to get a rough copy up and then we can iterate 🙂

**Comment by iamclaytonray** [comment]
Sorry for being so late on this. Totally dropped the ball. Umm... @cjoudrey - Can we add a task list to the original issue that includes which rules should be fixable?

**Comment by iamclaytonray** [comment]
I'll be working on this for at least today and hopefully more. I've never written a lint fixer so I'm in new waters. May take a bit but I'll do my best 😄

**Comment by cjoudrey** [comment]
@iamclaytonray I've never written something like this before either. 😁 I would have looked into [ESLint](https://github.com/eslint/eslint) and [Rubocop](https://github.com/bbatsov/rubocop).

Conceptually, I was thinking each rule would be able to optionally provide a function that the linter would use to _fix_ the issue. I think we'll want to make these changes by manipulating the AST at some point.

One challenge is that the AST we use during validation is the concatenation of all `.graphql` files. We may want to have a second step after validation called `fix` which would take the reported errors, parse the individual files causing the mistake, and have the rule fix that AST.

> Can we add a task list to the original issue that includes which rules should be fixable?

:+1: Done

**Comment by iamclaytonray** [comment]
That makes sense to me! Thanks for the kind words 😄 

I've been looking into TSLint and ESLint. Didn't think of Rubocop. I'll have to check that one out. Thanks!

**Comment by iamclaytonray** [comment]
So that I can check these off:

- [ ] EnumValuesAllCaps
- [ ] EnumValuesSortedAlphabetically
- [ ] FieldsAreCamelCased
- [ ] InputObjectValuesAreCamelCased
- [ ] TypesAreCapitalized

**Comment by iamclaytonray** [comment]
@cjoudrey - I'd like to integrate this fix feature (when it's done) with `graphql-cli` as well. Do you have any thoughts on that? I'm sure Johannes would love to chime in as well

**Comment by cjoudrey** [comment]
> I'd like to integrate this fix feature (when it's done) with graphql-cli as well.

@iamclaytonray yeah, that makes sense. You might be interested by this open issue: https://github.com/graphql-cli/graphql-cli/issues/14

At the moment [`graphql lint` does schema linting](https://github.com/graphql-cli/graphql-cli/blob/7b642f659b318fc64618c3d3b82886a725cd39f9/src/cmds/lint.ts), but I recall Johannes mentioning he wanted to have `graphql lint schema`. `graphql lint` would be for queries.

You might want to have that discussion with him in that issue to confirm that's still what he had in mind.

**Comment by iamclaytonray** [comment]
Sounds good!

**Comment by cjoudrey** [comment]
@iamclaytonray have you managed to make any progress on this by any chance? 😀😀

**Comment by iamclaytonray** [comment]
Sorry, no :/

I had a family emergency that's had me busy for the past week.

I actually just started picking this up again this morning.

**Comment by cjoudrey** [comment]
@iamclaytonray No need to be sorry! I was just curious. 😄 Hope all is well.

**Comment by iamclaytonray** [comment]
Hey @cjoudrey. I am still hacking away at this, a little bit at a time. I just got super slammed with work so I've been doing what I can in my off time. If anyone else in the community wants to help me out, I would definitely appreciate it 😄

**Comment by jlwt90** [comment]
@cjoudrey Hello, do you still want this feature? I think it might be helpful for us too :P

**Comment by cjoudrey** [comment]
> @cjoudrey Hello, do you still want this feature? I think it might be helpful for us too :P

Hi @jlwt90! 😄I still think this would be helpful. I'm not actively working on it though. PRs are welcome.

**Comment by danfoley** [comment]
@cjoudrey +1 can we please merge this? It seems like this project has been abandoned for a while. If the owner of the project is not working on it anymore can the ownership be at least shared with others who are working on the project so that a great PR like this can be merged and released?
